lr-cli/cortx_setup/commands/cluster/config/component.py

```diff
@@ -90,22 +92,32 @@ def run(self, component_group):
                         cortx_components[component_group],
                         stages=['config.config', 'config.init_mod']
                     )
-                self.logger.debug(f"Deploying cortx ha components on {primary}")
-                self._configure(
-                    ['ha.cortx-ha'],
-                    stages=[
-                        'config.post_install', 'config.prepare',
-                        'config.config', 'config.init_mod'],
-                    targets=primary
-                )
-                self.logger.debug(f"Deploying cortx ha components on {secondaries}")
-                self._configure(
-                    ['ha.cortx-ha'],
-                    stages=[
-                        'config.post_install', 'config.prepare',
-                        'config.config', 'config.init_mod'],
-                    targets=secondaries
-                )
+                if node_count > 1:
+                    self.logger.debug(f"Deploying cortx ha components on {primary}")
```

Comment:
The deployment block for primary can be kept out of "if" check and have that check only for secondary nodes. This way code block for primary will not be duplicated.